src/rss_etl.py

```diff
@@ -0,0 +1,175 @@
+        revisions_output_record = self.create_revisions_output_record(bill, bill_identifier)
+        if not revisions_output_record.has_existing_supabase_records(["revision_guid"]):
+            self.insert_new_record(revisions_output_record)
+            return True
+        return False
+
+    def create_and_attempt_to_insert_bill(self, bill, bill_identifier: BillIdentifier) -> bool:
+        bills_output_record = self.create_bill_output_record(bill, bill_identifier)
+        if not bills_output_record.has_existing_supabase_records(["legislative_id"]):
+            self.insert_new_record(bills_output_record)
+            return True
+        return False
+
+    # Method to obtain bill internal id from existing supabase row for a bill
+    def get_bill_internal_id(self, bill_identifier: BillIdentifier):
+        statement = self.supa_con.table("Bills").select("bill_internal_id").eq("legislative_id",
+                                                                               bill_identifier.bill_guid)
+        api_response = statement.execute()
+        return api_response.data[0]["bill_internal_id"]
+
+    def create_revisions_output_record(self, bill, bill_identifier: BillIdentifier):
+        revisions_output_record = OutputRecord(self.supa_con, "Revisions", REVISIONS_OUTPUT_DICT)
+        revisions_output_record.output_dict["bill_internal_id"] = self.get_bill_internal_id(bill_identifier)
+        revisions_output_record.output_dict["printer_no"] = bill_identifier.printer_number
+        revisions_output_record.output_dict["full_text_link"] = bill["link"]
+        revisions_output_record.output_dict["publication_date"] = bill["published"]
+        revisions_output_record.output_dict["revision_guid"] = bill["guid"]
+        revisions_output_record.output_dict["description"] = bill["description"]
+        return revisions_output_record
+
+    def create_bill_output_record(self, bill, bill_identifier: BillIdentifier):
+        bills_output_record = OutputRecord(self.supa_con, "Bills", BILL_OUTPUT_DICT)
+        bills_output_record.output_dict["bill_number"] = bill_identifier.bill_number
+        bills_output_record.output_dict["legislative_id"] = bill_identifier.bill_guid
+        bills_output_record.output_dict["legislative_session"] = bill_identifier.legislative_session
+        bills_output_record.output_dict["session_type"] = bill_identifier.session_type
+        bills_output_record.output_dict["chamber"] = bill_identifier.chamber
+        return bills_output_record
+
+    def get_bill_and_printer_no(self, title):
+        bp_regex = f"({self.leg_body}) (Bill|Resolution) (\d+) (Printer's Number) (\d+)"
+        result = re.search(bp_regex, title)
+        return result.group(3), result.group(5)
+
+    def insert_new_record(self, out_rec: OutputRecord):
+        self.supa_con.table(out_rec.table_name).insert(out_rec.output_dict).execute()
+
+
+def extract_from_rss_feed(supabase_connection, leg_bod, rss_feed):
```

Comment:
Generally, this line and everything else could just be kept in this file, and everything else could be helper files.